### PR TITLE
feat: Copy backup samples

### PR DESCRIPTION
**Pull request by asthamohta**
Adding the following samples for Copy Backup:
1. Copy Backup from source database
2. Use max expire time for backups
3. Get database options for copy backup

### CODE REVIEW

File: samples/snippets/src/main/java/com/example/spanner/CopyBackupSample.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+            Timestamp.ofTimeMicroseconds(
+                  TimeUnit.MICROSECONDS.convert(
+                          System.currentTimeMillis() + TimeUnit.DAYS.toMillis(14),
+                          TimeUnit.MILLISECONDS));
+    // Creates a copy of an existing backup.
+    Backup destinationBackup =
+          databaseAdminClient
+                  .newBackupBuilder(BackupId.of(projectId, instanceId, destinationBackupId))
+                  .setExpireTime(expireTime)
+                   .build();
```
Comment (olavloite) [inline samples/snippets/src/main/java/com/example/spanner/CopyBackupSample.java:86]: nit: fix indentation

File: samples/snippets/src/main/java/com/example/spanner/CopyBackupSample.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    System.out.println("Copying backup [" + destinationBackup.getId() + "]...");
+    OperationFuture<Backup, CopyBackupMetadata> op =
+          databaseAdminClient.copyBackup(
+                  BackupId.of(projectId, instanceId, sourceBackupId), destinationBackup);
+    try {
+      // Wait for the backup operation to complete.
+      destinationBackup = op.get();
+      System.out.println("Copied backup [" + destinationBackup.getId() + "]");
+    } catch (ExecutionException e) {
+      throw (SpannerException) e.getCause();
```
Comment (olavloite) [inline samples/snippets/src/main/java/com/example/spanner/CopyBackupSample.java:98]: nit: you can use `SpannerExceptionFactory.asSpannerException(e)`

File: samples/snippets/src/main/java/com/example/spanner/CopyBackupSample.java [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    } catch (InterruptedException e) {
+      throw SpannerExceptionFactory.propagateInterrupt(e);
+    }
+    // Reload the metadata of the backup from the server.
```
Comment (olavloite) [inline samples/snippets/src/main/java/com/example/spanner/CopyBackupSample.java:102]:
```suggestion
    // Load the metadata of the new backup from the server.
```

File: samples/snippets/src/main/java/com/example/spanner/SpannerSample.java [hunk 4]
```diff
@@ -1687,6 +1690,30 @@ static void listBackupOperations(InstanceAdminClient instanceAdminClient, Databa
...
       }
     }
+    // Get copy backup operations for the sample database.
+    filter =
+              String.format(
+                      "(metadata.@type:type.googleapis.com/"
+                              + "google.spanner.admin.database.v1.CopyBackupMetadata) "
+                              + "AND (metadata.source_backup:%s)",
+                      backupId.getName());
+    Page<Operation>  copyBackupOperations = instance.listBackupOperations(Options.filter(filter));
```
Comment (olavloite) [inline samples/snippets/src/main/java/com/example/spanner/SpannerSample.java:1700]: nit:
```suggestion
    Page<Operation> copyBackupOperations = instance.listBackupOperations(Options.filter(filter));
```

File: samples/snippets/src/test/java/com/example/spanner/SpannerSampleIT.java [hunk 5]
```diff
@@ -95,7 +95,7 @@ public static void setUp() throws Exception {
     key = String.format("projects/%s/locations/%s/keyRings/%s/cryptoKeys/%s",
         options.getProjectId(), keyLocation, keyRing, keyName);
   }
-
```
Comment (olavloite) [inline samples/snippets/src/test/java/com/example/spanner/SpannerSampleIT.java:98]: nit: remove these unrelated whitespace changes

**Review by olavloite — COMMENTED** [review]
There's no integration test for the `CopyBackup` sample. Is that intentional?

**Comment by asthamohta** [comment]
@olavloite yes, because I notices not all samples have an IT and this is a very long running function so decided to go with this

File: samples/snippets/src/test/java/com/example/spanner/SpannerSampleIT.java [hunk 6]
```diff
@@ -328,6 +328,9 @@ public void testSample() throws Exception {
               "Backup %s on database %s pending:",
               backupId.getName(),
               dbId.getName()));
+      assertThat(out).contains(
```
Comment (thiagotnunes) [inline samples/snippets/src/test/java/com/example/spanner/SpannerSampleIT.java:331]: nit: don't use `assertThat`, we should prefer to use `Assert.assert*` methods, such as `Assert.assertTrue("Out does not contain " + ..., out.contains(...))`

File: samples/snippets/src/main/java/com/example/spanner/CopyBackupSample.java [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+import com.google.cloud.spanner.SpannerExceptionFactory;
+import com.google.cloud.spanner.SpannerOptions;
+import com.google.spanner.admin.database.v1.CopyBackupMetadata;
+import java.time.LocalDateTime;
+import java.time.OffsetDateTime;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+public class CopyBackupSample {
+  public static void main(String[] args) {
```
Comment (thiagotnunes) [inline samples/snippets/src/main/java/com/example/spanner/CopyBackupSample.java:37]: I don't think any of the other single file samples have a main method, so we could remove this, no?

File: samples/snippets/src/main/java/com/example/spanner/CopyBackupSample.java [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+    // Creates a copy of an existing backup.
+    Backup destinationBackup =
+          databaseAdminClient
+                  .newBackupBuilder(BackupId.of(projectId, instanceId, destinationBackupId))
+                  .setExpireTime(expireTime)
+                   .build();
+
+    // Initiate the request which returns an OperationFuture.
+    System.out.println("Copying backup [" + destinationBackup.getId() + "]...");
+    OperationFuture<Backup, CopyBackupMetadata> op =
```
Comment (thiagotnunes) [inline samples/snippets/src/main/java/com/example/spanner/CopyBackupSample.java:90]: nit: please don't abbreviate variable names: `op` -> `operation`

File: samples/snippets/src/main/java/com/example/spanner/SpannerSample.java [hunk 9]
```diff
@@ -1659,21 +1660,23 @@ static void cancelCreateBackup(
   // [END spanner_cancel_backup_create]
 
   // [START spanner_list_backup_operations]
-  static void listBackupOperations(InstanceAdminClient instanceAdminClient, DatabaseId databaseId) {
+  static void listBackupOperations(
```
Comment (thiagotnunes) [inline samples/snippets/src/main/java/com/example/spanner/SpannerSample.java:1663]: Isn't this a breaking change? Maybe customers won't care because it is in the sample , but I don't know what the policy is here. Yoshi team might know

**Comment by ansh0l** [comment]
@asthamohta : can you update the PR to provide the description in the right format?

**Review by ansh0l — CHANGES_REQUESTED** [review]
Update the PR description

File: samples/snippets/src/main/java/com/example/spanner/SpannerSample.java (hunk 9 above)
Comment (asthamohta) [reply samples/snippets/src/main/java/com/example/spanner/SpannerSample.java:1662]: @olavloite pointed out that this is a non public method and hence can't be called so this will not be a breaking change

File: README.md [hunk 10]
```diff
@@ -41,28 +41,28 @@ If you are using Maven without BOM, add this to your dependencies:
...
 ```Groovy
-implementation 'com.google.cloud:google-cloud-spanner:6.23.0'
+implementation 'com.google.cloud:google-cloud-spanner:6.23.2'
 ```
 
 If you are using SBT, add this to your dependencies
 
 ```Scala
-libraryDependencies += "com.google.cloud" % "google-cloud-spanner" % "6.23.0"
+libraryDependencies += "com.google.cloud" % "google-cloud-spanner" % "6.23.2"
```
Comment (thiagotnunes) [inline README.md:65]: nit: I don't think these version updates should be part of this PR

File: google-cloud-spanner/pom.xml [hunk 11]
```diff
@@ -343,7 +343,7 @@
     <dependency>
       <groupId>org.openjdk.jmh</groupId>
       <artifactId>jmh-core</artifactId>
-      <version>1.34</version>
+      <version>1.35</version>
```
Comment (thiagotnunes) [inline google-cloud-spanner/pom.xml:346]: This should happen automatically no? Why do we need to update jmh here?

File: samples/install-without-bom/pom.xml [hunk 12]
```diff
@@ -32,7 +32,7 @@
     <dependency>
       <groupId>com.google.cloud</groupId>
       <artifactId>google-cloud-spanner</artifactId>
-      <version>6.21.2</version>
+      <version>6.23.2</version>
```
Comment (thiagotnunes) [inline samples/install-without-bom/pom.xml:35]: This is already part of this PR: https://github.com/googleapis/java-spanner/pull/1788/files

File: samples/snippets/pom.xml [hunk 13]
```diff
@@ -33,7 +33,7 @@
       <dependency>
         <groupId>com.google.cloud</groupId>
         <artifactId>libraries-bom</artifactId>
-        <version>25.0.0</version>
+        <version>25.1.0</version>
```
Comment (thiagotnunes) [inline samples/snippets/pom.xml:36]: This should happen automatically in another PR (from dependabot)

File: samples/snippets/src/main/java/com/example/spanner/SpannerSample.java (hunk 9 above)
Comment (thiagotnunes) [reply samples/snippets/src/main/java/com/example/spanner/SpannerSample.java:1663]: That is true, but we publish these samples in our public documentation, so we would be changing the signature. Anyway, if your team is ok with this, I am ok.

File: README.md (hunk 10 above)
Comment (asthamohta) [reply README.md:65]: So I didn't do this. After I raised the PR, the changes in this file came automatically after I raised the PR. I only changes the install-without-bom. I'll try merging the other PR with the version change and remove these files

Comment (thiagotnunes) [reply README.md:65]: Thanks!

File: samples/snippets/src/main/java/com/example/spanner/SpannerSample.java [hunk 14]
```diff
@@ -1840,13 +1867,19 @@ static void updateBackup(DatabaseAdminClient dbAdminClient, BackupId backupId) {
...
     System.out.println(String.format(
         "Updating expire time of backup [%s] to %s...",
         backupId.toString(),
         LocalDateTime.ofEpochSecond(
-            expireTime.getSeconds(),
-            expireTime.getNanos(),
-            OffsetDateTime.now().getOffset()).toString()));
+          expireTime.getSeconds(),
+          expireTime.getNanos(),
+          OffsetDateTime.now().getOffset()).toString()));
```
Comment (olavloite) [inline samples/snippets/src/main/java/com/example/spanner/SpannerSample.java:1882]: super nit: the `.toString()` is superfluous

**Review by olavloite — APPROVED** [review]
